cmake/OmimHelpers.cmake

```diff
@@ -33,7 +33,7 @@ function(omim_add_executable executable)
 
   # Enable warnings for all our binaries.
   target_compile_options(${executable} PRIVATE ${OMIM_WARNING_FLAGS})
-  target_include_directories(${executable} PRIVATE ${OMIM_INCLUDE_DIRS})
+  target_include_directories(${executable} SYSTEM PRIVATE ${OMIM_INCLUDE_DIRS})
```

Comment:
Generally, it doesn't look right to mark some 3p lib includes as system ones. Are there any possible side effects from this change?

Reply:
Well, it seems it depends on compiler. 

> If the SYSTEM option is given, the compiler will be told the directories are meant as system include directories on some platforms. Signalling this setting might achieve effects such as the compiler skipping warnings, or these fixed-install system files not being considered in dependency calculations - see compiler docs.

https://cmake.org/cmake/help/latest/command/include_directories.html

From my own experience I've never experienced any problems with this myself. I've heard about issue that compiler/cmake may assume those headers never change(i.e. it may break your workflow if you change something in "system" dependency and then iteratively rebuilds the project), but not sure this can end up with some problem exactly in this place(after all `OMIM_INCLUDE_DIRS ` contains Boost headers, not sure we will ever use Boost this way)

But in general IMO it is the only way to avoid multiple clang-tidy alerts we won't have any real chance to fix(imagine if someone forgot to add `reserve` to vector somewhere in Boost header, it is not critical problem, but it will block us from building the project).

Reply:
Ok. Please double-check about cmake behavior, because we periodically update the boost version and other deps. It would be a pity to waste a lot of time by debugging it later.